pkg/cmd/adm/unregister_member_test.go

```diff
@@ -36,9 +39,55 @@ func TestUnregisterMemberWhenAnswerIsY(t *testing.T) {
 	assert.Contains(t, term.Output(), "Delete Member cluster stated above from the Host cluster?")
 	assert.Contains(t, term.Output(), "The deletion of the Toolchain member cluster from the Host cluster has been triggered")
 	assert.NotContains(t, term.Output(), "cool-token")
-
 	AssertDeploymentHasReplicas(t, fakeClient, hostDeploymentName, 1)
-	assert.Equal(t, 2, numberOfUpdateCalls)
+}
+
+func TestUnregisterMemberWhenRestartError(t *testing.T) {
+	// given
+	toolchainCluster := NewToolchainCluster(ToolchainClusterName("member-cool-server.com"))
+	hostDeploymentName := test.NamespacedName("toolchain-host-operator", "host-operator-controller-manager")
+	deployment := newDeployment(hostDeploymentName, 1)
+	deployment.Labels = map[string]string{"kubesaw-control-plane": "kubesaw-controller-manager"}
+
+	newClient, fakeClient := NewFakeClients(t, toolchainCluster, deployment)
+	numberOfUpdateCalls := 0
+	fakeClient.MockUpdate = whenDeploymentThenUpdated(t, fakeClient, hostDeploymentName, 1, &numberOfUpdateCalls)
```

Comment:
correct me if I'm wrong, but there is no need to create the Deployment anymore.
```suggestion
	newClient, fakeClient := NewFakeClients(t, toolchainCluster)
```
the same applies to the other cases as well